src/storage/wal/wal.cpp

```diff
@@ -34,13 +38,13 @@ WAL::WAL(const std::string& directory, bool readOnly, BufferManager& bufferManag
 WAL::~WAL() {}
 
 void WAL::logBeginTransaction() {
-    lock_t lck{mtx};
+    std::unique_lock<std::mutex> lck{mtx};
```

Comment:
why are we changing this?

Reply:
I removed the typedef `lock_t` in header file, which I don't think it's necessary.